src/Datadog.Trace/Vendors/StatsdClient/Statsd.cs

```diff
@@ -236,7 +237,7 @@ public Statsd(IStatsdUDP udp, IRandomGenerator randomGenerator, IStopWatchFactor
             RandomGenerator = randomGenerator;
             _prefix = prefix;
 
-            string entityId = Environment.GetEnvironmentVariable(StatsdConfig.DD_ENTITY_ID_ENV_VAR);
+            string entityId = EnvironmentHelpers.GetEnvironmentVariable(StatsdConfig.DD_ENTITY_ID_ENV_VAR);
```

Comment:
This code is copied from [dogstatsd-csharp-client](https://github.com/DataDog/dogstatsd-csharp-client) and any changes will be overwritten by our [UpdateVendors](https://github.com/DataDog/dd-trace-dotnet/tree/master/tools/UpdateVendors) tool. We may have to add the try/catch in our code when creating the `Statsd` instance.

Reply:
We can also add a transform in the vendoring tool to automatically swap `Environment.GetEnvironmentVariable` with `EnvironmentHelpers.GetEnvironmentVariable`

Reply:
The creation of the statsd client is already wrapped in a try/catch block so that'll do for now I think, given how rare those errors should be. We can update the vendoring tool if that becomes an issue.